<!--
	Launch file for Baxter's hand-eye calibration
	
	After launching this file...
		
	SoCS, UoG. Gerardo Aragon-Camarasa & Aamir Khan. 09/2016
-->

<launch>
	
	<!-- Configuration file for camera calibration -->
    <param name="/glasgow_calibration/target" 	     value="opencv" />
    <param name="/glasgow_calibration/marker_width"  value="0.03" />
    <param name="/glasgow_calibration/marker_size_x" value="5" />
    <param name="/glasgow_calibration/marker_size_y" value="7" />
    <param name="/glasgow_calibration/max_error"     value="2.0" />
    
    <!-- Where to save stuff -->
    <param name="/glasgow_calibration/outputImageDir" value="$(find calibration_glasgow)/calibrations/Images/" />
    <param name="/glasgow_calibration/outputCalibDir" value="$(find calibration_glasgow)/calibrations" />
    
    <!-- Saving? 0: Only calib stuff is saved, 1: save processed images in hard disk plus calib stuff, 2: Save everything-->
    <param name="/glasgow_calibration/save_mode" value="2" type="int"/>
    <!-- Scale ratio in order to reduce the image while finding the inner corners-->
    <param name="/glasgow_calibration/resize_imgs_factor" value="1.0"/>

    <!-- Where to save the gripper to camera calibration file -->
    <param name="/glasgow_calibration/gHc_calibration_file" value="$(find calibration_glasgow)/" />
    
	<!-- Calibration service node -->
	<node pkg="calibration_glasgow" type="calibration_services" name="calibration_services" output="screen">
	</node>

<!--====================================baxter在moveit中的部分====================================================-->
<!--
	<arg name="moveit_sensor_manager" default="baxter" />
	<arg name="kinect" default="false" />
	<arg name="xtion" default="true" />
	<arg name="camera_link_pose" default="0.0 0.0 0.0 0.0 0.0 0.0"/>
	<include file="$(find baxter_moveit_config)/launch/$(arg moveit_sensor_manager)_moveit_sensor_manager.launch" >
		<arg name="kinect" value="$(arg kinect)" />
		<arg name="xtion" value="$(arg xtion)" />
		<arg name="camera_link_pose" default="$(arg camera_link_pose)"/>
	</include>
-->

	<!-- Messages node -->
	<arg name="robot_ip" default="192.168.1.123"/>
	<include file="$(find kinect2_bridge)/launch/kinect2_bridge.launch">
    </include>
	<include file="$(find ur_modern_driver)/launch/ur3_bringup.launch">
		<arg name="robot_ip" value="$(arg robot_ip)"/>
    </include>
	<!-- 
	<node pkg="calibration_glasgow" type="robot_trans_pub" name="robot_helper" output="screen">
	</node>
	-->


	<!--
	<node pkg="twodto3d" type="twodto3d" name="twodto3d" output="screen">
	</node>
	<node pkg="twodto3d" type="transform_pub" name="transform_pub" output="screen">
	</node>
	-->
	<!-- For debug -->
	<param name="/glasgow_calibration/debug" type="bool" value="False" />
	
</launch>

